segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ZKSegmentContainerMonitor.java

```diff
@@ -255,8 +255,7 @@ private CompletableFuture<ContainerHandle> startContainer(int containerId) {
         log.info("Starting Container {}.", containerId);
         this.pendingTasks.add(containerId);
         try {
-            return CompletableFuture.runAsync(() -> Exceptions.handleInterrupted(parallelContainerStartsSemaphore::acquire))
-                    .thenCompose(v -> this.registry.startContainer(containerId, INIT_TIMEOUT_PER_CONTAINER))
+            return this.parallelContainerStartsSemaphore.run(() -> this.registry.startContainer(containerId, INIT_TIMEOUT_PER_CONTAINER), 1, false)
```

Comment:
what executor this will be running on? is this running on Fork Join pool? if yes is it okay?